File: array_adt/exercises/countDuplicate.cc
// check how many times duplicate values are occurred in sorted array

#include <iostream>
using namespace std;

int main(){
	int array[] = {3,6,8,8,10,12,15,15,15,20};
	int length = sizeof(array)/sizeof(int);
	int j = 0;

	// O(n)
	for (int i = 0; i < length-1; ++i){
		if (array[i] == array[i+1]){
			j = i; // 2
			while (array[j] == array[i]){
				j++;
			}
			cout << array[i] << " is occurred " << (j-i) << " times" << endl;
			i = j - 1;
		}
	}
}

File: array_adt/exercises/countDuplicate.py
# find the duplicate elements in an sorted array using hash table

# space and time is O(n)
def duplicate(array, array2):
	for i in range(len(array)):
		array2[array[i]] += 1

	for i in range(len(array2)):
		if (array2[i] > 1):
			print(f'{i} is occurred {array2[i]} times')


if __name__ == '__main__':
	array = [3,6,8,8,10,12,15,15,15,20]
	array2 = 21*[0]
	duplicate(array, array2)